<?xml version="1.0" encoding="utf-8"?>

<launch>

  <arg name="nodelet_manager" default="sort_ros_nodelet_manager" />

  <!-- <node pkg="rosbag" type="play" name="player" output="screen" args="-s 90 /media/disk1/orcvio/mit_rosbags/phoenix_stack/outdoor_demo_test_2020-11-04-15-48-26.bag"/> -->
  <node pkg="rosbag" type="play" name="player" output="screen" args="/media/disk1/orcvio/mit_rosbags/phoenix_stack/medfield_ft2_2020-11-06-11-58-28.bag"/>
  <!-- <node pkg="rosbag" type="play" name="player" output="screen" args="/media/disk1/orcvio/mit_rosbags/phoenix_stack/outdoor_demo_test_2020-11-04-15-48-26.bag"/> -->

  <arg name="name" default="acl_jackal"/>

  <!-- <include file="$(find darknet_ros)/launch/darknet_ros.launch" >
    <arg name="image" value="/husky/camera/image_raw" />
  </include> -->

  <!-- convert mask rcnn detection node  -->
  <arg name="algorithm" default="maskrcnn"
       doc="The network to use for object detection (maskrcnn|yolov4)"/>

  <node if="$(eval algorithm == 'maskrcnn')"  ns="name" 
        name="orcvio_bridge"
        pkg="darknet_ros_bridge"
        type="darknet_ros_bridge.py"
        output="screen"
        >
    <remap from="~ira_dets_in" to="/$(arg name)/detection_out_topic"/>
    <remap from="~darknet_bounding_boxes_out" to="/$(arg name)/darknet_ros/bounding_boxes"/>
  </node>


  <node pkg="nodelet" type="nodelet" name="SortRos"
    args="standalone sort_ros/SortRos"
    output="screen">
    <param name="image_topic" value="/$(arg name)/forward/color/image_raw" />
    <param name="bbox_topic" value="/$(arg name)/darknet_ros/bounding_boxes" />
    <param name="tracked_bbox_topic" value="/sort_ros/tracked_bounding_boxes" />
    
    <param name="max_age" value="100" />
    <param name="min_hits" value="5" />
    <param name="iou_threshold" value="0" />
    <param name="use_centroid_dist_flag" type="bool" value="true" />
    <param name="centroid_dist_threshold" value="200" />
  </node>

  <node name="republish1" type="republish" pkg="image_transport" output="screen" args="compressed in:=/$(arg name)/forward/color/image_raw raw out:=/$(arg name)/forward/color/image_raw" />

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find sort_ros)/launch/display_dcist.rviz" /> 
  <!-- <node pkg="rqt_image_view" type="rqt_image_view" name="rqt_image_view_det" args=" /SortRos/detection_image/compressed"/> -->

</launch>